core/storage/btree.rs

```diff
@@ -5541,8 +5674,8 @@ impl CursorTrait for BTreeCursor {
                         // should be safe as contents is not a leaf page
                         let right_most_pointer = contents.rightmost_pointer().unwrap();
                         self.stack.advance();
-                        let (mem_page, c) = self.read_page(right_most_pointer as i64)?;
-                        self.stack.push(mem_page);
+                        let (child, c) = return_if_io!(self.read_page(right_most_pointer as i64));
```

Comment:
How does this work? If we yield on IO here, then `child` won't be pushed to stack. If we don't yield on IO here, then is `c` ever `Some`?